client/components/import/import.js

```diff
@@ -11,48 +11,105 @@ const ImportPopup = BlazeComponent.extendComponent({
     return 'importPopup';
   },
 
-  events() {
-    return [{
-      'submit': (evt) => {
-        evt.preventDefault();
-        const dataJson = $(evt.currentTarget).find('.js-import-json').val();
-        let dataObject;
-        try {
-          dataObject = JSON.parse(dataJson);
-        } catch (e) {
-          this.setError('error-json-malformed');
-          return;
-        }
-        Meteor.call(this.getMethodName(), dataObject, this.getAdditionalData(),
-          (error, response) => {
-            if (error) {
-              this.setError(error.error);
-            } else {
-              Filter.addException(response);
-              this.onFinish(response);
-            }
-          }
-        );
-      },
-    }];
+  jsonText() {
+    return Session.get('import.text');
+  },
+
+  membersMapping() {
+    return Session.get('import.membersToMap');
   },
 
   onCreated() {
     this.error = new ReactiveVar('');
+    this.dataToImport = '';
+  },
+
+  onFinish() {
+    Popup.close();
+  },
+
+  onShowMapping(evt) {
+    this._storeText(evt);
+    Popup.open('mapMembers')(evt);
+  },
+
+  _storeText() {
+    const dataJson = this.$('.js-import-json').val();
+    Session.set('import.text', dataJson);
+    return dataJson;
+  },
+
+  onSubmit(evt){
```

Comment:
This submit function looks a bit complex to me, could you move the mapping logic in a separate method?